### PR TITLE
Dispose MemoryStream in ConvertFrom method of CursorConverter

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Looks like that a MemoryStream used in ConvertFrom method should be disposed.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3476)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs [hunk 1]
```diff
@@ -55,25 +55,23 @@ public override bool CanConvertTo(ITypeDescriptorContext context, Type destinati
...
                     }
                 }
             }
 
-            if (value is byte[])
+            if (value is byte[] bytes)
             {
-                MemoryStream ms = new MemoryStream((byte[])value);
+                using MemoryStream ms = new MemoryStream(bytes);
                 return new Cursor(ms);
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs:75]: The `Cursor(Stream)` constructor just reads the data into a byte array:
```cs
public Cursor(Stream stream)
{
    if (stream == null)
    {
        throw new ArgumentNullException(nameof(stream));
    }

    int length = checked((int)stream.Length);
    _cursorData = new byte[length];
    stream.Read(_cursorData, 0, length);
    LoadPicture(
        new Ole32.GPStream(new MemoryStream(_cursorData)),
        nameof(stream));
}
```
 
We could feasibly create an internal ctor that takes `byte[]`? Not sure if this is performance critical

File: src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs [hunk 2]
```diff
@@ -55,25 +55,23 @@ public override bool CanConvertTo(ITypeDescriptorContext context, Type destinati
         /// </summary>
         public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
         {
-            if (value is string)
+            if (value is string s)
             {
-                string text = ((string)value).Trim();
+                string text = s.Trim();
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs:60]: Nit: we could initialise the string as a `Span`, call `Trim` on it (https://docs.microsoft.com/en-us/dotnet/api/system.memoryextensions.trim?view=netcore-3.1) and then use `Span.Equals` (https://docs.microsoft.com/en-us/dotnet/api/system.memoryextensions.equals?view=netcore-3.1)

Not sure if this is perf critical or even how likely it is that the string has trailing/leading whitespace

File: src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs (hunk 1 above)
Comment (weltkante) [reply src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs:75]: The quoted code is actually broken in multiple ways
- if the stream position isn't at zero the buffer would be allocated too large
- `Read` is allowed to read less then `length` bytes (e.g. decompressed or network streams)

This code should do a `stream.CopyTo(MemoryStream)` to let it do the heavy lifting.

Note the current code doesn't throw, it just over-allocates and then reads less bytes than requested, which is fine if the cursor data is still represented completely at the current position.

Should we fix it in this PR or create a separate issue?

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs:75]: Thank you for the assessment.
If you don't mind, please create a new issue for this.

Comment (weltkante) [reply src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs:75]: done in #3492

as far as the original comment is concerned

> We could feasibly create an internal ctor that takes byte[] and route the Stream ctor to the byte[] ctor?

I'd make a public constructor taking a `Span<byte>` that would be useful for everyone, but needs API design discussions, created #3493

> Not sure if this is performance critical

unlikely

**Pull request merged**